pkg/agent/plugin/nodeattestor/aws/iid.go

```diff
@@ -109,6 +118,54 @@ func fetchMetadata(endpoint string) (*caws.IIDAttestationData, error) {
 	}, nil
 }
 
+func getMetadataDoc(client *imds.Client) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), fetchMetadataDocTimeout)
+	defer cancel()
+	res, err := client.GetDynamicData(ctx, &imds.GetDynamicDataInput{
+		Path: docPath,
+	})
+
+	if err != nil {
+		return "", err
+	}
```

Comment:
nitpick: prefer the error handling to immediately follow (keeps things grouped nicely)
```suggestion
	res, err := client.GetDynamicData(ctx, &imds.GetDynamicDataInput{
		Path: docPath,
	})
	if err != nil {
		return "", err
	}
```